scripts/generate_network_config_bookworm.py

```diff
@@ -0,0 +1,138 @@
+from piaware_config import get_standard_config_group, ConfigGroup
+from uuid import uuid4
+import subprocess
+import os
+
+SYS_CON_DIR = "/etc/NetworkManager/system-connections"
+
+def format_dns(dns_string: str) -> str:
+    names = dns_string.strip(" \n\t")
+    names = names.split(" ")
+    formatted = ";".join(names) + ";"
+    return formatted
+
+def check_address_and_netmask_set(network_type: str, config: ConfigGroup):
+    if config.get(f"{network_type}-address") is None:
+        raise ValueError(f"{network_type}-type was set to static but address was not set")
+    if config.get(f"{network_type}-netmask") is None:
+        raise ValueError(f"{network_type}-type was set to static but netmask was not set")
+
+def configure_static_network(network_type: str, config: ConfigGroup):
+    check_address_and_netmask_set(network_type, config)
+
+    ip = ""
+    static_ip = config.get(f"{network_type}-address") + "/" + config.get(f"{network_type}-netmask")
+    gateway = config.get(f"{network_type}-gateway")
+    if gateway is not None:
+        static_ip += f",{gateway}"
+    ip += f"address1={static_ip}\n"
+
+    name_servers = config.get(f"{network_type}-nameservers")
+    if name_servers is not None:
+        ip += f"dns={format_dns(name_servers)}\n"
+    return ip
```

Comment:
Since what you're doing here is building a list of lines, it might be more obvious to literally build a list of lines, then str.join() the list at the end.